core/src/main/java/org/primefaces/extensions/component/clockpicker/ClockPickerRenderer.java

```diff
@@ -0,0 +1,200 @@
+
+        writer.startElement("div", clockPicker);
+        writer.writeAttribute("class", ClockPicker.CONTAINER_CLASS, null);
+
+        writer.startElement("input", null);
+        writer.writeAttribute("id", inputId, null);
+        writer.writeAttribute("name", inputId, null);
+        writer.writeAttribute("type", "text", null);
+        writer.writeAttribute("class", "ui-inputfield ui-widget ui-state-default ui-corner-all", null);
+        writer.writeAttribute("size", 5, null);
+        writer.writeAttribute("maxlength", 5, null);
+
+        if (LangUtils.isNotBlank(value)) {
+            writer.writeAttribute("value", value, null);
+        }
+
+        writer.endElement("input");
+
+        writer.startElement("span", null);
+        writer.writeAttribute("class", "input-group-addon", null);
+        writer.startElement("span", null);
+        writer.writeAttribute("class", "glyphicon glyphicon-time", null);
+        writer.endElement("span");
+        writer.endElement("span");
+        writer.endElement("div");
+    }
+
+    private void encodeScript(final FacesContext context, final ClockPicker clockPicker, final String value) throws IOException {
+        final WidgetBuilder wb = getWidgetBuilder(context);
+        final String clientId = clockPicker.getClientId(context);
+
+        wb.init("ExtClockPicker", clockPicker.resolveWidgetVar(), clientId);
+        encodeClientBehaviors(context, clockPicker);
+
+        wb.attr("placement", clockPicker.getPlacement());
+        wb.attr("align", clockPicker.getAlign());
+        wb.attr("donetext", clockPicker.getDonetext());
+        wb.attr("autoclose", clockPicker.getAutoclose());
+        wb.attr("vibrate", clockPicker.getVibrate());
+
+        wb.finish();
+    }
+
+    protected static String getValueAsString(final FacesContext context, final ClockPicker clockPicker) {
+        final Object submittedValue = clockPicker.getSubmittedValue();
+        if (submittedValue != null) {
+            return submittedValue.toString();
+        }
+
+        final Object value = clockPicker.getValue();
+        if (value == null) {
+            return null;
+        }
+        else {
+            if (clockPicker.getConverter() != null) {
+                // convert via registered converter
+                return clockPicker.getConverter().getAsString(context, clockPicker, value);
+            }
+            else {
+                // use built-in converter
+                SimpleDateFormat timeFormat;
+                timeFormat = new SimpleDateFormat("HH:mm", clockPicker.calculateLocale());
+
+                return timeFormat.format(value);
+            }
+        }
+    }
+
+    @Override
+    public Object getConvertedValue(FacesContext context, UIComponent component,
+                Object value) throws ConverterException {
+        final ClockPicker clockPicker = (ClockPicker) component;
+        String submittedValue = (String) value;
+        SimpleDateFormat format = null;
```

Comment:
I would have used `LocalTime` as the default type. It's better suited for time only and easier to work with.